managed/services/victoriametrics/prometheus.go

```diff
@@ -209,9 +213,55 @@ func AddInternalServicesToScrape(cfg *config.Config, s models.MetricsResolutions
 		scrapeConfigForAlertmanager(s.MR),
 		scrapeConfigForGrafana(s.MR),
 		scrapeConfigForPMMManaged(s.MR),
-		scrapeConfigForQANAPI2(s.MR))
+		scrapeConfigForQANAPI2(s.MR),
+		scrapeConfigForServerPMMAgent(s.MR))
 	// TODO Refactor to remove boolean positional parameter when Prometheus is removed
 	if dbaas {
 		cfg.ScrapeConfigs = append(cfg.ScrapeConfigs, scrapeConfigForDBaaSController(s.MR))
 	}
 }
+
+// AddPmmAgentScrapeConfig adds pmm-agent metrics to scrape targets.
+func AddPmmAgentScrapeConfig(cfg *config.Config, l *logrus.Entry, q *reform.Querier, s *models.MetricsResolutions, pmmAgentID *string) error {
```

Comment:
can we return a scrape config here and add it in the method above it should make the code more consistent